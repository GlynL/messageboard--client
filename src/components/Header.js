import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import "../styles/header.css";

const Header = props => {
  function renderLinks() {
    if (props.authenticated) {
      return (
        <div className="header__nav-group">
          <Link className="header__nav-item" to="/new-board">
            New Board
          </Link>
          <Link className="header__nav-item" to="/signout">
            Sign out
          </Link>
        </div>
      );
    }
    return (
      <div className="header__nav-group">
        <Link className="header__nav-item" to="/signup">
          Sign Up
        </Link>
        <Link className="header__nav-item" to="/signin">
          Sign In
        </Link>
      </div>
    );
  }

  return (
    <nav className="header__nav">
      <Link className="header__nav-item" to="/">
        MessageBoard
      </Link>
      {renderLinks()}
    </nav>
  );
};

const mapStateToProps = state => ({ authenticated: state.auth.authenticated });

export default connect(mapStateToProps)(Header);
