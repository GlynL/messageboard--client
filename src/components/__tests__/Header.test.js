import React from "react";
import { shallow } from "enzyme";
import { Header } from "../Header";

it("matches unauthenticated snapshot", () => {
  const wrapped = shallow(<Header authenticated={false} />);
  expect(wrapped).toMatchSnapshot();
});

it("matches authenticated snapshot", () => {
  const wrapped = shallow(<Header authenticated={true} />);
  expect(wrapped).toMatchSnapshot();
});
