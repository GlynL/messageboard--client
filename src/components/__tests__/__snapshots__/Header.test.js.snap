// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches authenticated snapshot 1`] = `
<nav
  className="header__nav"
>
  <Link
    className="header__nav-item"
    replace={false}
    to="/"
  >
    MessageBoard
  </Link>
  <div
    className="header__nav-group"
  >
    <Link
      className="header__nav-item"
      replace={false}
      to="/signout"
    >
      Sign out
    </Link>
  </div>
</nav>
`;

exports[`matches unauthenticated snapshot 1`] = `
<nav
  className="header__nav"
>
  <Link
    className="header__nav-item"
    replace={false}
    to="/"
  >
    MessageBoard
  </Link>
  <div
    className="header__nav-group"
  >
    <Link
      className="header__nav-item"
      replace={false}
      to="/signup"
    >
      Sign Up
    </Link>
    <Link
      className="header__nav-item"
      replace={false}
      to="/signin"
    >
      Sign In
    </Link>
  </div>
</nav>
`;
